{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/settings.py

```diff
@@ -1,124 +1,208 @@
+BASE_DIR = Path(__file__).absolute().parent.parent
+PROJECT_NAME = "{{cookiecutter.project_name}}"
+CONFIGURATION = environ["DJANGO_CONFIGURATION"]
+CONFIG_DIR = environ.get("DJANGO_CONFIG_DIR")
+SECRET_DIR = environ.get("DJANGO_SECRET_DIR")
 
-# Detect if we are running tests.  Is this really the best way?
-IN_TESTS = 'test' in argv
+# Detect if we are running tests.
+IN_TESTS = environ.get("RUNNING_TESTS")
 
 
-class RedisCache(object):
-    CACHES = {
-        'default': {
-            'BACKEND': 'django_redis.cache.RedisCache',
-            'LOCATION': 'redis://{}:{}/1'.format(getenv('REDIS_SERVICE_HOST', '127.0.0.1'), getenv('REDIS_SERVICE_PORT', 6379)),
-            'KEY_PREFIX': '{}_'.format(PROJECT_ENVIRONMENT_SLUG),
-            'OPTIONS': {
-                'CLIENT_CLASS': 'django_redis.client.DefaultClient',
-                'PARSER_CLASS': 'redis.connection.HiredisParser',
-                # You may want this. See https://niwinz.github.io/django-redis/latest/#_memcached_exceptions_behavior
-                # 'IGNORE_EXCEPTIONS': True, # see
-            }
-        }
-    }
-    SESSION_ENGINE = 'django.contrib.sessions.backends.cache'
-    SESSION_CACHE_ALIAS = 'default'
+def get_env(name, default=None, required=False, cast=str):
+    """
+    Get an environment variable
+    Arguments:
+        name (str): Name of environment variable
+        default (Any): default value
+        required (bool): If True, raises an ImproperlyConfigured error if not defined
+        cast (Callable): function to call with extracted string value.
+            Not applied to defaults.
+    """
+
+    def _lookup(self):
+        value = environ.get(name)
+
+        if value is None and default is not None:
+            return default
+
+        if value is None and required:
+            raise ValueError(f"{name} not found in env")
+
+        return cast(value)
+
+    return property(_lookup)
+
+
+def get_secret(name, cast=str):
+    """
+    Get a secret from disk
+    Secrets should be available as the content of `<SECRET_DIR>/<name>`
+    All secrets are required
+    Arguments:
+        name (str): Name of environment variable
+        cast (Callable): function to call on extracted string value
+    """
+
+    # We don't want this to be called unless we're in a configuration which uses it
+    def _lookup(self):
+        if not SECRET_DIR:
+            raise ValueError(
+                f"Secret {name} not found: DJANGO_SECRET_DIR not set in env"
+            )
+
+        file = Path(SECRET_DIR) / name
+        if not file.exists():
+            raise ValueError(f"Secret {file} not found")
+
+        value = file.read_text().strip()
+        return cast(value)
+
+    return property(_lookup)
+
+
+def csv_to_list(value):
+    """
+    Convert a comma separated list of values into a list.
+    Convenience function for use with get_env() and get_secret() ``cast`` argument.
+    """
+    if value is None:
+        return []
+    return value.split(",")
```

Comment:
If we want to use these three fns, would it be worth splitting them out into their own project to simplify tests and maintainability?